CorsixTH/Lua/staff_profile.lua

```diff
@@ -143,8 +143,20 @@ function StaffProfile:randomise(month)
 end
 
 function StaffProfile:randomiseOrganical()
-  -- TODO: Randomise a letter from the current language's alphabet.
-  self.name = "" -- string.char(string.byte"A" + math.random(0, 25)) .. ". "
+  local letters = tostring(our_concat(_S.humanoid_name_starts) .. our_concat(_S.humanoid_name_ends)):sub(33)
+  -- If UTF8 then make a table of the letters and pick a random one
+  if letters:find("([%z\1-\127\194-\244][\128-\191]*)") then
+    local initials = {}
+    for uchar in string.gmatch(letters,
+      "([%z\1-\127\194-\244][\128-\191]*)") do
+      initials[#initials + 1] = uchar
+    end
+    self.name = initials[math.random(1, #initials)] .. ". "
+  else
+    local num = math.random(1, letters:len())
+    self.name = letters:sub(num, num) .. ". "
+  end
```

Comment:
Seems a bit expensive to compute all possible letters each time, since it's always the same unless you change language. So I propose that you make a cache of available initials.

Initialize the cache when you arrive here for the first time (ie when the cache is empty). I'd always simply assume we have utf-8 to simplify the code.
If you want to be resistant against change of language, you can store the relevant `_S` strings too in the cache, and compare them with the current value (and rebuild the cache if the cache is empty or the `_S` strings don't match).